lib/internal/modules/cjs/loader.js

```diff
@@ -532,20 +496,20 @@ function applyExports(basePath, expansion) {
 // 1. name/.*
 // 2. @scope/name/.*
 const EXPORTS_PATTERN = /^((?:@[^/\\%]+\/)?[^./\\%][^/\\%]*)(\/.*)?$/;
-function resolveExports(nmPath, request, absoluteRequest) {
+function resolveExports(nmPath, request) {
   // The implementation's behavior is meant to mirror resolution in ESM.
-  if (!absoluteRequest) {
-    const [, name, expansion = ''] =
-      StringPrototypeMatch(request, EXPORTS_PATTERN) || [];
-    if (!name) {
-      return path.resolve(nmPath, request);
-    }
-
-    const basePath = path.resolve(nmPath, name);
-    return applyExports(basePath, expansion);
+  const [, name, expansion = ''] =
```

Comment:
using array destructuring means it will break if Symbol.iterator is deleted off of Array.prototype. this one has a lot of places in the codebase to fix, tho, so it’s probably fine to skip for now.